### PR TITLE
fix: Take previously connected git applications accessibility into consideration to accurately calculate private repo count 

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> When we are connecting a application to git-repo we are not taking into account the fact that users can change the accessibility for the previously connected application from the remote repo, this leads to the incorrect number of currently connected private repo count. This PR makes sure to check the accessibility for all the private repo in an org and update the application git data if user alters the accessibility from remote.

Fixes https://github.com/appsmithorg/appsmith/issues/10684

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Junit TC
> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 1]
```diff
@@ -411,7 +411,7 @@ private String getReason() {
                                     .flatMap(application -> gitCloudServicesUtils.getPrivateRepoLimitForOrg(application.getOrganizationId(), false)
                                             .flatMap(limitCount -> {
                                                 //get git connected apps count from db
-                                                return applicationService.getGitConnectedApplicationCount(application.getOrganizationId())
+                                                return applicationService.getGitConnectedApplicationsCountWithPrivateRepoByOrgId(application.getOrganizationId())
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:414]: IIUC, say there are 3 connected Application(a1, a2, a3-public-repo ). Now if i connect a 4th repo (private) and then make the a3 private repo then commit+push operation will go through right?
We should call the `getGitConnectedApplicationsWithPrivateRepoByOrgId` in commit operation right?

Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:414]: In the commit application, we check if the repo is public and also if the user has toggled accessibility to private, if yes then again follow the same flow to check the available repo count limit before making the commit. On the other hand for private repo we commit the application directly.

Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:414]: We have the same check present in connectFlow as well. We need the count only if the repo is private. Else any of the above conditions won't be present in the flow. So i don't understand how is the scenario i mentioned before is handled?

**Review by AnaghHegde — APPROVED** [review]
LGTM

**Pull request merged**